webhook/mutation/pod_mutator.go

```diff
@@ -204,264 +224,485 @@ func isWellKnownWorkload(ownerRef metav1.OwnerReference) bool {
+	var rsp admission.Response
+	endpointGenerator := dtingestendpoint.NewEndpointSecretGenerator(m.client, m.apiReader, m.namespace, log)
+
+	var endpointSecret corev1.Secret
+	if err := m.apiReader.Get(ctx, client.ObjectKey{Name: dtingestendpoint.SecretEndpointName, Namespace: ns.Name}, &endpointSecret); k8serrors.IsNotFound(err) {
+		if _, err := endpointGenerator.GenerateForNamespace(ctx, dkName, ns.Name); err != nil {
+			log.Error(err, "failed to create the data-ingest endpoint secret before pod injection")
+			rsp = admission.Errored(http.StatusBadRequest, err)
+			return nil, &rsp
+		}
+	} else if err != nil {
+		log.Error(err, "failed to query the data-ingest endpoint secret before pod injection")
+		rsp = admission.Errored(http.StatusBadRequest, err)
+		return nil, &rsp
+	}
+
+	dataIngestFields, err := endpointGenerator.PrepareFields(ctx, &dk)
+	if err != nil {
+		log.Error(err, "failed to query the data-ingest endpoint secret before pod injection")
+		rsp = admission.Errored(http.StatusBadRequest, err)
+		return nil, &rsp
+	}
+	return dataIngestFields, nil
+}
+
+func (m *podMutator) getDk(ctx context.Context, req admission.Request, dkName string) (dynatracev1beta1.DynaKube, *admission.Response) {
+	var rsp admission.Response
+	var dk dynatracev1beta1.DynaKube
+	if err := m.client.Get(ctx, client.ObjectKey{Name: dkName, Namespace: m.namespace}, &dk); k8serrors.IsNotFound(err) {
+		template := "namespace '%s' is assigned to DynaKube instance '%s' but doesn't exist"
+		m.recorder.Eventf(
+			&dynatracev1beta1.DynaKube{ObjectMeta: metav1.ObjectMeta{Name: "placeholder", Namespace: m.namespace}},
+			corev1.EventTypeWarning,
+			missingDynakubeEvent,
+			template, req.Namespace, dkName)
+		rsp = admission.Errored(http.StatusBadRequest, fmt.Errorf(
+			template, req.Namespace, dkName))
+		return dynatracev1beta1.DynaKube{}, &rsp
+	} else if err != nil {
+		rsp = admission.Errored(http.StatusInternalServerError, err)
+		return dynatracev1beta1.DynaKube{}, &rsp
+	}
+	return dk, nil
+}
+
+func (m *podMutator) ensureInitSecret(ctx context.Context, ns corev1.Namespace, dk dynatracev1beta1.DynaKube) *admission.Response {
+	var initSecret corev1.Secret
+	var rsp admission.Response
+
+	if err := m.apiReader.Get(ctx, client.ObjectKey{Name: dtwebhook.SecretConfigName, Namespace: ns.Name}, &initSecret); k8serrors.IsNotFound(err) {
+		if _, err := initgeneration.NewInitGenerator(m.client, m.apiReader, m.namespace, log).GenerateForNamespace(ctx, dk, ns.Name); err != nil {
+			log.Error(err, "Failed to create the init secret before pod injection")
+			rsp = admission.Errored(http.StatusBadRequest, err)
+			return &rsp
+		}
+	} else if err != nil {
+		log.Error(err, "failed to query the init secret before pod injection")
+		rsp = admission.Errored(http.StatusBadRequest, err)
+		return &rsp
+	}
+	return nil
+}
+
+func (m *podMutator) CreateInjectionInfo(pod *corev1.Pod) *InjectionInfo {
```

Comment:
would fit perfectly in `injection_info.go`, it doesn't need to be a method of `podMutator` as it never uses`m`.
Nitpick: maybe calling it `NewInjectionInfoForPod` (or something) would help understandability when its no longer a method of `podMutator`